cli/src/commands/prev.rs

```diff
@@ -78,25 +77,18 @@ pub(crate) fn cmd_prev(
             .heads()
             .contains(current_wc_id);
     let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
-    let start_id = if edit {
-        current_wc_id
+    let amount = if current_wc.is_discardable() && !args.edit {
+        // If the current working copy is discardable and the user didn't
+        // specify --edit, then we move one commit further back. This is so that
+        // the new empty working commit is created on the parent of the current
+        // on (or further back depending on the value of args.amount). If we
+        // didn't do this, then we wouldn't move far enough.
+        args.amount + 1
```

Comment:
In my mental mode, the current position is determined when I ran `jj new`/`jj edit`, not when I made some changes on the working-copy commit. In colocated repo, it's basically where `HEAD@git` points.

Reply:
I agree with Yuya, it clearly shouldn't matter if a commit is _discardable_ or not.

Reply:
@yuja @martinvonz @PhilipMetzger 
Of course commands which move `@` determine the current position. That's self evident. I think what you're saying is that you're opposed to treating discardable commits differently. Let's table that discussion for now.

The problem with `jj prev` is that there's no way to go from this:
```
jj log -T 'separate(" ",change_id.short(), description)'
@  umspkqwnpqvx add file2
◉  nprzlkrnkpuu add file1
◉  zzzzzzzzzzzz
```

To this:
```
jj log -T 'separate(" ",change_id.short(), description)'
@  okxltnqrlpsr
│ ◉  umspkqwnpqvx add file2
├─╯
◉  nprzlkrnkpuu add file1
◉  zzzzzzzzzzzz
```

If I specify an offset of one, `jj prev`, using somewhat convoluted logic, actually moves back two revisions instead of one. It is impossible to move back by one without specifying `--edit`, which is a problem.

I changed this PR so that the user specified offset is never changed and discardable commits aren't treated any differently. This means that running `jj prev` on a discardable commits gets you a new discardable commit, and as a result `@` appears not to have changed. This is consistent though, and you can move back by two if you specify an offset of 2.

I will move my ergonomics change to a separate PR. I'm fine if we don't submit that, but I think this PR should be submitted because it is **clearly a bug** that I cannot create a new empty commit on the parent of the current commit using `jj prev`.

Reply:
I know a lot of people (like @hooper) consider the changes in the working-copy commit as significantly different from changes in the working copy in other VCS. I see them as equivalent. If that's how we want others to think of them, we might need to document that better. Alternatively, if we seem them as significantly different, we should document how they're different.

Reply:
> Of course commands which move `@` determine the current position. That's self evident. I think what you're saying is that you're opposed to treating discardable commits differently. Let's table that discussion for now.

But the whole point you've made everywhere is about that, so don't chicken out.

Reply:
I'm not sure I consider the working copy changes to be different.

But I do consider discardable commits to be different and I disagree that they should be treated equally because doing so is often not practical from a usability perspective. That's why discardable commits are discarded when @ moves. I have to admit that I really don't understand why some people are adamant that discardable commits are equal to all other commits. It seems like an argument about theoretical purity to me that gets in the way of usability. I'm mostly interested in making the CLI easy to use.

Reply:
> > Of course commands which move `@` determine the current position. That's self evident. I think what you're saying is that you're opposed to treating discardable commits differently. Let's table that discussion for now.
> 
> But the whole point you've made everywhere is about that, so don't chicken out.

`@` is literally the current position, so anything that changes it determines the current position. It's a bit tautological.

My stance (which you disagree with) is that we should, in some cases, adjust how commands that move `@` work if `@` is a discardable commit. I haven't changed my opinion, I'd just like to focus this PR on fixing the buggy behavior of `jj prev`. I also said I'm fine to drop my ergonomics suggestion since the same behavior can be accomplished with `jj prev 2` after this PR. Let's call that compromise instead of chickening out :)

Reply:
> I'm mostly interested in making the CLI easy to use.

While treating `jj` as `hg` right? You still think of `@` as _not_ being attached to a parent and that the working copy is actually not a commit in the graph sense. While this seems _consistent_ from that perspective, we never cared about that anyway, atleast not in the `jj` I know.

Reply:
> I cannot create a new empty commit on the parent of the current commit using jj prev.

If I want to do that with `jj prev`, I would call it `jj prev 0`.

Reply:
I'm thinking of `jj prev` as `jj new @-` where the minus is multiplied by the offset. In other words, move `@` back by the offset and then run `jj new`. Or if `--edit` is supplied, don't run `jj new`.

So the offset is how far `@` is moved. And `--edit` controls whether a new empty commit is created or not.

I'd expect that `jj prev 0` would not move `@`, so it would create a new commit on top of the current one I suppose.

Reply:
> I'm thinking of `jj prev` as `jj new @-` where the minus is multiplied by the offset. In other words, move `@` back by the offset and then run `jj new`. Or if `--edit` is supplied, don't run `jj new`.

I'm not against adding such mode. I think one of the current problem is that the `--edit` option is somewhat overloaded to describe both "the current working-copy commit is editing" and "continue editing the destination commit".

Anyway, I insist that it's confusing to change the behavior depending on `wc.is_discardable()`. In the "currently-editing, but not edit the destination" mode, `jj prev` will be equivalent to `jj new @-` no matter if `@` is empty or not.

Reply:
I think if I had tried `jj next` sooner I would have been a lot less confused about how these functions work, so that's on me.

It seems like if `--edit` is not used or implied, then next and prev move `@` back by 1 and then move it according to the user specified offset and create a new commit. That behavior is consistent and useful if you're usually editing a leaf commit. I think we can say the movement in this case is relative to `@-`, which I did not realize before.

I will say that this behavior is a little bit surprising to me for `jj prev`, but I see that it's necessary for `jj next` to be useful and for `jj prev` to be its inverse. **I'd like to retract this PR and my suggestion to change the movement behavior.**

If you don't mind I will probably send a PR to tweak the documentation a little bit based on this discussion so that others with the same misunderstanding might be less confused.